scripts/cleanup_failed_registrations.sh

```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+TEMPDIR=`mktemp -d`
+trap "rm -rf $TEMPDIR" EXIT
```

Comment:
Very interesting. Why do we need to create a temporary directory in the first place? I ask more out of curiosity of shell scripting than anything.